src/cypress-api/support.ts

```diff
@@ -0,0 +1,90 @@
+import { snapshot } from '@chromaui/rrweb-snapshot';
+import type { ResourceArchive } from '../resource-archive';
+import './commands';
+
+const setupNetworkListener = () => {
+  let pageUrl: URL | null = null;
+  // these "archive" and "manualSnapshots" variables will be available before, during, and after the test,
+  // then cleaned up before the next test is run
+  // (see https://docs.cypress.io/guides/core-concepts/variables-and-aliases#Aliases)
+  // @ts-expect-error will fix when Cypress has its own package
+  cy.wrap({}).as('archive');
+  // @ts-expect-error will fix when Cypress has its own package
+  cy.wrap([]).as('manualSnapshots');
+
+  // since we don't know where the user will navigate, we'll archive whatever domain they're on first.
+  // @ts-expect-error will fix when Cypress has its own package
+  cy.intercept(`**/*`, (req) => {
+    // don't archive the page itself -- we'll do that with rrweb
+    // TODO: See if this will work for both slash and not slash endings or if we have to do same "first URL visited" stuff
+    if (!pageUrl) {
+      pageUrl = new URL(req.url);
+      return;
+    }
+
+    const url = new URL(req.url);
+    if (url.origin !== pageUrl.origin) {
+      return;
+    }
+
+    // cached (304) responses don't provide a body, so we need to make sure cache is blown away
+    // (https://glebbahmutov.com/blog/cypress-intercept-problems/#cached-response)
+    // https://github.com/cypress-io/cypress/issues/15680
+    delete req.headers['if-none-match'];
+    // I added this since some css files still are cached... not sure if this is great
+    // (https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/304)
+    delete req.headers['if-modified-since'];
+    // @ts-expect-error will fix when Cypress has its own package
+    req.continue((response) => {
+      // @ts-expect-error will fix when Cypress has its own package
+      cy.get('@archive')
+        .then((archive: ResourceArchive) => {
+          const updatedArchive = {
+            ...archive,
+            [response.url]: {
+              statusCode: response.statusCode,
+              statusText: response.statusMessage,
+              body: response.body,
+            },
+          };
+          return updatedArchive;
+        })
+        .as('archive');
+    });
+  });
```

Comment:
I have a future [ticket](https://linear.app/chromaui/issue/AP-3791/cypress-feature-parity-with-playwrights-network-listening) to make sure we get parity with all the things Playwright handles with network requests, but let me know if anything sticks out here as glaringly missing.

I'm hopeful that by intercepting requests on the front-end we may be able to keep this simple!

Reply:
The only thing I can think of is maybe handling redirects?